### PR TITLE
finished event creation

### PR DESCRIPTION
**Pull request by brandonfujii**
made locations place id unique so there is no location redundancy in the locations table, finished events services for create event

### CODE REVIEW

File: src/requests/EventCreationRequest.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+let Request = require('./Request')
+
+let bodyRequired = ['name', 'userId', 'startDate', 'endDate', 'location']
+let bodyValidations = {
```
Comment (SamirJoshi) [inline src/requests/EventCreationRequest.js]: Are descriptions not required in the event creation request, what about invitees?

File: src/models/Event.js [hunk 2]
```diff
@@ -13,15 +13,15 @@ let Event = Model.extend({
   validations: {
     name: ['required', 'string'],
     description: 'string',
-    userId: ['required', 'integer'],
+    user_id: ['required', 'integer'],
     location_id: ['required', 'integer'],
-    start_date: ['required', 'date'],
-    end_date: ['required', 'date']
+    start_date: ['required', 'string'],
```
Comment (SamirJoshi) [inline src/models/Event.js]: are we not doing datetime objects here for sorting

File: src/requests/EventCreationRequest.js (hunk 1 above)
Comment (brandonfujii) [inline src/requests/EventCreationRequest.js]: I don't think adding a description should be mandatory but it is definitely allowed in an event creation request. I'm going to add invitees right now. I wanted to get a first iteration of the model/request going.

Comment (SamirJoshi) [inline src/requests/EventCreationRequest.js]: 👍

File: src/models/Event.js (hunk 2 above)
Comment (brandonfujii) [inline src/models/Event.js]: What datetime objects were you thinking of using ?

Comment (brandonfujii) [inline src/models/Event.js]: I think we can query our db with something like `SELECT * FROM events WHERE end_date  >= CURDATE();`, which is valid with a date string like we have now.

Comment (brandonfujii) [inline src/models/Event.js]: @Franklify @nkgopaul What do you guys think?

**Pull request merged**